### ISSUE TITLE
Layering of tools not recorded in State

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/geometric-optics-basics#22]
**Device**
Dell
**OS**
Win 11
**Browser**
Firefox
**Problem Description**
For https://github.com/phetsims/qa/issues/799
If you pull multiple tools out of the tool box in State, the layering of those tools isn't necessarily the same in the bottom sim as in the top sim.
**Visuals**
<img width="519" alt="rulerlayering" src="https://user-images.githubusercontent.com/41024075/164311168-b202fa8e-70f8-4c70-9451-fec00fe6ca6c.png">

**Comment by pixelzoom** [comment]
All tools are in one "layer". When you press on a tool, or remove it from the toolbox, it moves to the front of that layer.   This is common behavor for tools.

@zepumph @samreid is this something that you recommend making "stateful". If so, how has it been addressed in other PhET-iO sims?

**Comment by samreid** [comment]
We introduced an IOType to support this, called `IndexedNodeIO`, which is described as:

> IO Type for Nodes that can save their own index (if phetioState: true).  Can be used to customize z-order
or layout order.

I haven't tested it recently, so if it doesn't work out-of-the-box, please notify us.

**Comment by zepumph** [comment]
Yes agreed, I would start by looking at the usage in ComboBoxListItemNode, since that is the main reason it was created.

**Comment by pixelzoom** [comment]
I'm going to change the label on this from "bug" to "enhancement", since it's not a feature that was requested.

**Comment by pixelzoom** [comment]
Migrating this issue from geometric-optics-basics to geometric-optics, because it's relevant to both sims, and the implementation will be in geometric-optics.
